src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -154,10 +164,34 @@ protected ExtensionsRunner(Extension extension) throws IOException {
         // save custom transport actions
         this.transportActions = new TransportActions(extension.getActionsMap());
 
-        ThreadPool threadPool = new ThreadPool(this.getSettings());
+        Guice.createInjector(b -> {
+            b.bind(ExtensionsRunner.class).toInstance(this);
+            b.bind(Extension.class).toInstance(extension);
+
+            b.bind(NamedXContentRegistry.class).toInstance(getNamedXContentRegistry().getRegistry());
+            b.bind(ThreadPool.class).toInstance(getThreadPool());
```

Comment:
Need to understand this. Do we need ThreadPool in SDK?

Reply:
This is SDK's own thread pool.  

Do we need it?  No.  Do we want a single common thread pool that the whole extension can share if it so chooses, to balance its resources?  Why not?

Reply:
Also from a PR standpoint, this does not change the previous behavior of a threadpool being created (line 157) and passed to the extension via create components.  Just changes the syntax of how an extension accesses that object.

Reply:
Makes sense. Do we need to bind ThreadPool then, as it is specific to the SDK?

Reply:
In theory we might make this object available for in-proc extensions to be passed as a param.  I don't know, but we can always easily remove it before GA.